src/specs/01_basic/qe81_non_party_transactions.spec

```diff
@@ -160,7 +160,7 @@ Transactions, regardless if it succeeds or not, sent by non-party node must not
 * Contract `C1`("parentContractC1_14")'s `get()` function execution in "Node2" returns "0"
 * Contract `C2`("childContractC2_12")'s `get()` function execution in "Node1" returns "30"
 * Contract `C2`("childContractC2_12")'s `get()` function execution in "Node4" returns "0"
-* Contract `C2`("childContractC2_12")'s `get()` function execution in "Node2" returns "0"
+* Fail to execute contract `C2`("childContractC2_12")'s `get()` function in "Node2" with error "execution reverted"
```

Comment:
why we need to change this step and not `Contract `C2`("childContractC2_12")'s `get()` function execution in "Node4" returns "0"` which does the same thing but for another Node?

Reply:
Because they don't do the same behaviour.

## cases
### case 1 
node4, privy to C1, but not to C2, calls C2. `eth_call` returns a `ContractCallException` with [a message `Empty value (0x)`](https://github.com/ConsenSys/quorum-acceptance-tests/tree/master/src/main/java/com/quorum/gauge/services/NestedContractService.java#L147-L149) (like the default behaviour in geth, from what I understood) because this contract is not available to node4. The test `readC2Value` function returns 0 in that case.

### case 2
node2, privy to C2, but not to C1, calls C1. `eth_call` returns a `ContractCallException` without the `Empty value (0x)` because the C2 contract call C1.get(), but that fails because node2 is not privy to C1. The test `readC2Value` function throws a `RuntimeException` in that case. So we need to catch this exception instead.

## solutions

- change readC2Value behaviour
- create a new step function

I went with the second because it seems more behaviourly correct : the value is not 0, it doesn't have access, that's all.